listenbrainz_spark/ftp/download.py

```diff
@@ -94,21 +94,68 @@ def download_spark_dump_and_get_path(self, directory, dump_id, ftp_dump_dir, dum
         current_app.logger.info('Done. Total time: {:.2f} sec'.format(time() - t0))
         return dest_path
 
-    def download_msid_mbid_mapping(self, directory, mapping_dump_id=None):
-        """ Download msid_mbid_mapping to dir passed as an argument.
+    def get_mapping_dump_name_of_given_type(self, dump, mapping_type=None):
+        """ Get list of given mapping type dirs.
+
+            Args:
+                dump: list of dumps in the current working directory.
+                mapping_type (str): mapping type.
+
+            Returns:
+                mapping: list of mapping dump names of given type in the current working directory.
+        """
+        if mapping_type is None:
+            err_msg = 'Please provide a valid mapping type from the following:' \
+                      '\nmsid-mbid-mapping\nmsid-mbid-mapping-with-text\nmsid-mbid-mapping-matchable\n'
+            raise MissingMappingTypeException(err_msg)
+
+        mapping = list()
+        for mapping_name in dump:
+            mapping_pattern = '{}-\\d+-\\d+(.tar.bz2)$'.format(mapping_type)
+
+            if re.match(mapping_pattern, mapping_name):
+                mapping.append(mapping_name)
+
+        if len(mapping) == 0:
+            err_msg = '{} type mapping not found'.format(mapping_type)
+            raise DumpNotFoundException(err_msg)
+
+        return mapping
+
+    def get_latest_mapping(self, mapping):
+        """ Get latest mapping name.
+
+            Args:
+                mapping: list of mapping dump names.
+
+            Returns:
+                latest_mapping (str): latest mapping dump name.
+        """
+        # sort the mappings on timestamp
+        sorted_mapping = sorted(mapping, key=lambda x: int(x.split('-')[-2] + x.split('-')[-1].split('.')[0]))
```

Comment:
This is a bit confusing -- you have a good regexp up above -- why not use that to parse the filename? Also, you should sort on the date string, not on the timestamp. The timestamp is only there to provide disambiguation in case we ever produce more than 1 dump in any given day.

Reply:
ummmm, so if we produce more than one dump in any given day, we would want to download the latest mapping of the two (of greater datetime)?

Reply:
Yes

Reply:
So we should sort on date and timestamp both?

Reply:
Yes

Reply:
Yeah. Was a bit confused by the original comment. Thanks :))